util/defaulting/defaulting.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaulting
+
+import (
+	"testing"
+
+	"github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// DefaultingValidator interface is for objects that define both defaulting
+// and validating webhooks.
+type DefaultingValidator interface {
+	admission.Defaulter
+	admission.Validator
+}
+
+// ValidateDefaultingTest returns a new testing function to be used in tests to
+// make sure defaulting webhooks also pass validation tests on create,
+// update and delete.
+func ValidateDefaultingTest(object DefaultingValidator) func(*testing.T) {
```

Comment:
Super nit: should this be `DefaultValidateTest` (it is more consistent with what the test is actually do and with the `DefaultingValidator` input type)

Reply:
have renamed.